sp/src/game/shared/mapbase/matchers.cpp

```diff
@@ -5,18 +5,20 @@
 // $NoKeywords: $
 //=============================================================================
 
+// Must be included before cbase.h, otherwise some macros break the Linux build.
+#include <regex>
```

Comment:
I changed it, I think it should work now.
I'll rebase my changes to squash the fixup!s if it now compiles without changes on your end.

Reply:
> Perhaps #ifdef __GNUC__ could instead use #define/#unfdef to disable the macros which are causing issues?

I got heaps of errors, but it was actually just `min` and `max` that cause problems. I'm `#undef`ining them now and including `minmax.h` after, just in case one expects the macros to be there. Should work now for both platforms!

Reply:
This compiles for me now. I personally would prefer this code to be within an `#ifdef __GNUC__` block since it isn't necessary on Win32 and it's made clear that it's for a specific platform, but just doing it like this doesn't seem to be causing any issues.